sig-scalability/slos/pod_startup_latency.md

```diff
@@ -4,29 +4,43 @@
+| __WIP__ | Startup latency of schedulable<sup>[1](#footnote1)</sup> stateful<sup>[3](#footnote3)</sup> pods, excluding time to pull images, run init containers and provision volumes (in delayed binding mode), measured from pod creation timestamp to when all its containers are reported as started and observed via watch, measured as 99th percentile over last 5 minutes | In default Kubernetes installation, 99th percentile per cluster-day <= X where X depends on storage provider |
 
-<a name="footnote1">[1\]</a>A `stateless pod` is defined as a pod that doesn't
+<a name="footnote1">[1\]</a>By schedulable pod we mean a pod that can be
+scheduled in the cluster without causing any preemption.
+
+<a name="footnote2">[2\]</a>A `stateless pod` is defined as a pod that doesn't
 mount volumes with sources other than secrets, config maps, downward API and
 empty dir.
 
-<a name="footnote2">[2\]</a>By schedulable pod we mean a pod that can be
-scheduled in the cluster without causing any preemption.
+<a name="footnote3">[2\]</a>A `stateful pod` is defined as a pod that mounts
+at least one volume with sources other than secrets, config maps, downward API
+and empty dir.
 
 ### User stories
 - As a user of vanilla Kubernetes, I want some guarantee how quickly my pods
 will be started.
 
 ### Other notes
-- Only schedulable and stateless pods contribute to the SLI:
+- Only schedulable pods contribute to the SLIs:
   - If there is no space in the cluster to place the pod, there is not much
     we can do about it (it is task for Cluster Autoscaler which should have
     separate SLIs/SLOs).
   - If placing a pod requires preempting other pods, that may heavily depend
     on the application (e.g. on their graceful termination period). We don't
     want that to contribute to this SLI.
-  - Mounting disks required by non-stateless pods may potentially also require
-    non-negligible time, not fully dependent on Kubernetes.
+- We are explicitly splitting stateless and stateful pods from each other:
+  - Starting a stateful pod requires attaching and mounting volumes, that
+    takes non-negligible amount of time and doesn't even fully depend on
+    Kubernetes. However, even though it depends on chosen storage provider,
+    it isn't application specific, thus we make that part of the SLI
+    (though the exact SLO threshold may depend on chosen storage provider).
+  - We also explicitly exclude time to provision a volume (in delayed volume
+    binding mode), even though that also only depends on storage provider,
+    not on the application itself. However, the time it takes to provision
+    a volume may be significant so initially we decided to exclude that from
+    the SLI. We will revisit that later.
```

Comment:
Another way you can think of it: Volume provisioning is a bootstrapping operation in the lifetime of a stateful pod.  You do volume provisioning only once at the beginning, not everytime a pod is created.

Reply:
I really like this POV. Added it to the text and removed todo to revisit that later.